test_runner/src/main/kotlin/ftl/args/ArgsHelper.kt

```diff
@@ -150,4 +172,24 @@ object ArgsHelper {
         // Allow users control over projectId by checking using Google's logic first before falling back to JSON.
         return ServiceOptions.getDefaultProjectId() ?: serviceAccountProjectId()
     }
+
+    // https://stackoverflow.com/a/2821201/2450315
+    private val envRegex = Pattern.compile("\\$([a-zA-Z_]+[a-zA-Z0-9_]*)")
+    private fun substituteEnvVars(text: String): String {
+        val buffer = StringBuffer()
+        val matcher = envRegex.matcher(text)
+        while (matcher.find()) {
+            val envName = matcher.group(1)
+            val envValue = System.getenv(envName) ?: ""
```

Comment:
Should we raise an error/warning here? Silently ignoring un-found env variables seems like a bug.

Reply:
this is standard behavior in Bash, that's how environment variables work sadly.

Reply:
discussed on slack, printing a warning makes sense.